### PR TITLE
Register working!

### PR DESCRIPTION
**Pull request by Fivvee0**

### CODE REVIEW

File: crates/client/src/account/register.rs [hunk 1]
```diff
@@ -41,10 +49,56 @@ pub fn register() -> Html {
...
+                type="text"
             />
             <button type="submit">{ "Register" }</button>
         </form>
         </>
     }
 }
+
+fn hash_password(password: &str) -> String {
+    let mut hasher = Sha256::new();
```
Comment (Vixenka) [inline crates/client/src/account/register.rs]: Sha256 is not secure now for storing passwords, better use Argon2 for better security
https://github.com/ArlekinTeam/ArlekinClient2023/blob/49ac627836bca0db3dd178baa3c3f5be988d9fb7/frontend/src/account/login.rs#L107C1-L113C23

File: crates/client/src/account/register.rs [hunk 2]
```diff
@@ -41,10 +49,56 @@ pub fn register() -> Html {
...
+    hasher.update(password);
```
Comment (Vixenka) [inline crates/client/src/account/register.rs]: Missing salt

File: crates/client/src/account/register.rs [hunk 3]
```diff
@@ -41,10 +49,55 @@ pub fn register() -> Html {
...
+    email: String,
+    password_hash: String,
+}
+
+async fn register_in_backend(
+    hashed_password: &str,
+    username: &str,
+    email: &str,
+) -> Result<(), reqwest::Error> {
+    let user = User {
```
Comment (Vixenka) [inline crates/client/src/account/register.rs]: Nothing is wrong here, but serde have json! macro which can be used in cases like this :)

File: crates/client/src/account/register.rs [hunk 4]
```diff
@@ -1,24 +1,32 @@
...
-        let password_input = get_by_id("password");
-
-        console::log_1(
-            &format!(
-                "Email: {}, Password: {}, Username: {}",
-                email_input, password_input, username_input
-            )
-            .into(),
-        );
+            let _ = register_in_backend(&hashed_password, &username_input, &email_input).await;
```
Comment (Vixenka) [inline crates/client/src/account/register.rs]: When function will return error then this error will be ignored, you don't proceed it in any way

**Review by Vixenka — CHANGES_REQUESTED** [review]
I see you wanted to create this fast, and this is fine. Everything can be improved in future merges.
But beyond this problems which I described in other comments, your solution doesn't work properly with CORS policy, and have very poor abstraction on making request to the API. When you or somebody else will be create e.g. login page, the most of that code will be cloned, I think you can get some inspiration basing on that what I made for old client [here](https://github.com/ArlekinTeam/ArlekinClient2023/blob/master/frontend/src/api/mod.rs). And remember you don't must do that in this merge request!

File: crates/client/src/account/register.rs (hunk 1 above)
Comment (Fivvee0) [reply crates/client/src/account/register.rs]: I will implement this later

File: crates/client/src/account/register.rs (hunk 3 above)
Comment (Fivvee0) [reply crates/client/src/account/register.rs]: thanks

File: crates/client/src/account/register.rs (hunk 4 above)
Comment (Fivvee0) [reply crates/client/src/account/register.rs]: I fixed

**Review by Vixenka — APPROVED** [review]

**Pull request merged**